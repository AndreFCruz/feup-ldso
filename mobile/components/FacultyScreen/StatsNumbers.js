import React from 'react';
import {View} from 'react-native';
import {Text } from 'native-base';
import PropTypes from 'prop-types';
import Dash from 'react-native-dash';

class StatisticsNumbers extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {course, students} = this.props;

        return (
            <View style={styles.main}>
                <View style={styles.section}>
                    <Text style={styles.numbers}>{course[0]}</Text>
                    <Text style={styles.text}>{course[1]}</Text>
                </View>
                <View style={styles.section}>
                    <Dash dashColor='white' dashGap={5} style={styles.dash} />
                </View>
                <View style={styles.section}>
                    <Text style={styles.numbers}>{students[0]}</Text>
                    <Text style={styles.text}>{students[1]}</Text>
                </View>
            </View>
        );
    }
}

StatisticsNumbers.propTypes = {
    course: PropTypes.array,
    students: PropTypes.array,
};

export default StatisticsNumbers ;

const styles = {
    main: {
        padding:20,
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#1c1c1c'
    },

    section: {
        padding: 3,
        flex:1,
        alignItems: 'center',
        justifyContent: 'center'
    },

    dash: {
        width:1,
        height:50,
        flexDirection:'column'
    },

    numbers: {
        fontFamily: 'Quicksand_regular',
        fontSize: 30,
        color: 'white',
        paddingBottom: 5
    },

    text: {
        color: 'white',
        fontFamily: 'Quicksand_bold',
        fontSize: 25
    }
};