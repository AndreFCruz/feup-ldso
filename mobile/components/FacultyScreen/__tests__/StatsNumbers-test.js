import React from 'react';
import StatsNumbers  from '../StatsNumbers';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
    const course = [10, 'Masters'];
    const students = [500, 'Students'];
    const wrapper = shallow(<StatsNumbers course={course} students={students}/>,
        {
            disableLifecycleMethods: true
        });
    return wrapper;
}

describe('Rendering', () => {
    it('Render component', () => {
        const wrapper = setup();
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
