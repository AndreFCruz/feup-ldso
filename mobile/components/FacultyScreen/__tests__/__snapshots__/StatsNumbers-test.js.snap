// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Render component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "padding": 53.5,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontFamily": "Quicksand_regular",
          "fontSize": 19,
          "paddingBottom": 13.5,
        }
      }
    >
      10
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans_regular",
          "fontSize": 12.5,
          "lineHeight": 13,
          "textAlign": "center",
        }
      }
    >
      Masters
    </Styled(Text)>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 4,
      }
    }
  >
    <_class
      dashColor="white"
      dashGap={5}
      style={
        Object {
          "flexDirection": "column",
          "height": 64,
          "width": 1,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontFamily": "Quicksand_regular",
          "fontSize": 19,
          "paddingBottom": 13.5,
        }
      }
    >
      500
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans_regular",
          "fontSize": 12.5,
          "lineHeight": 13,
          "textAlign": "center",
        }
      }
    >
      Students
    </Styled(Text)>
  </Component>
</View>
`;
